src/modules/item/sagas.ts

```diff
@@ -294,6 +361,22 @@ export function* itemSaga(legacyBuilder: LegacyBuilderAPI, builder: BuilderClien
     }
   }
 
+  function* handleDeleteItemSuccess(action: DeleteItemSuccessAction) {
+    const { item } = action.payload
+    const collectionId = item.collectionId!
+    const location: ReturnType<typeof getLocation> = yield select(getLocation)
+    if (location.pathname === locations.thirdPartyCollectionDetail(collectionId)) {
+      const paginationData: ItemPaginationData = yield select(getPaginationData, collectionId)
+      const { currentPage, limit, ids } = paginationData
+      const shouldGoToPreviousPage = currentPage > 1 && ids.length === 1 && ids[0] === item.id
+      if (shouldGoToPreviousPage) {
+        yield put(push(locations.thirdPartyCollectionDetail(collectionId, { page: currentPage - 1 })))
+      } else {
+        yield put(fetchCollectionItemsRequest(collectionId, { page: currentPage, limit }))
```

Comment:
Yeah, and what's the problem in that scenario? Sorry I didn't get it.
The the current code, `shouldGoToPreviousPage` will be `false`, so it will just re-fetch the same page (the 2nd one) and the user will see [4,6,7].

Reply:
Remember that we're not storing all the pages, we only save the page we asked. 
If the user is in 2nd page  looking at [4,5,6] and deletes the 5, the page 2 will be re-fetched and it will show `[4,6,7]`. So what we will have at that moment is that the paginated items are [4,6,7]. We don't know about the next page.
If the user moves to page 3 (by clicking the button), the page will be fetched (even if it was seen before, it always gets re-fetched when moving across pages) and so the server will reply with `[8,9,10]`.